### ISSUE TITLE
Query String in URL returns 404 when forward query toggle is off

### ISSUE THREAD

**Issue by Rhoyo** [skybrud/Skybrud.Umbraco.Redirects#187]
### Which version of Skybrud Redirects are you using? (Please write the *exact* version, example: 4.0.8)

4.0.8

### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

10.1.4

### Bug description

<img width="773" alt="Screenshot 2023-10-04 at 4 31 52 PM" src="https://github.com/skybrud/Skybrud.Umbraco.Redirects/assets/63175403/c618a586-4913-4bee-9529-601913378f96">

I have the forward query string toggled to disabled. However, when I curl and add a query string at the end it returns a 404, serving my custom 404 page. Not a 301 response. It seems it acts as it should when no query is added.

<img width="583" alt="Screenshot 2023-10-04 at 4 33 46 PM" src="https://github.com/skybrud/Skybrud.Umbraco.Redirects/assets/63175403/46258912-2863-40a3-8f98-c0fe2a15210a">

<img width="529" alt="Screenshot 2023-10-04 at 4 33 25 PM" src="https://github.com/skybrud/Skybrud.Umbraco.Redirects/assets/63175403/2840ca70-1de5-41e0-aaab-b62828a1f9ff">

When I toggle the forward query string and add a query string to my link it acts as it should, with and without a query string.
<img width="588" alt="Screenshot 2023-10-04 at 4 36 04 PM" src="https://github.com/skybrud/Skybrud.Umbraco.Redirects/assets/63175403/5a593e68-469e-4a43-9859-be199ffb74e5">
<img width="495" alt="Screenshot 2023-10-04 at 4 37 07 PM" src="https://github.com/skybrud/Skybrud.Umbraco.Redirects/assets/63175403/a4aad5fb-b89d-4574-9645-7a05bd9f76f8">

If I may ask, how come it returns a 404 when the forward query toggle is turned off and a query string is added? Is this something that could be accounted for or perhaps not? I was thinking more on the lines that the package would handle the query string, clip it off and return the url without the query string as a 301 response when the forward query toggle is off. For example: localhost:44388/test?q=test would be localhost:44388/test-page/ with a 301 status code instead of a 404. Thanks again @abjerner

**Comment by Rockerby** [comment]
I came across this today, but actually looks like this is intended behaviour. Some sites would want `/test?x=y` and `/test` to be different pages altogether. In my experience I feel as though the default behaviour should be to have **Forward query string** on, allowing users to turn it off.

What I think would be great is some default settings we could change; luckily enough there's something discussed in [i151](https://github.com/skybrud/Skybrud.Umbraco.Redirects/issues/151) that fits.

**Comment by abjerner** [comment]
Like @Rockerby mentions, this is the indented behavior. It's something we settled on when starting the package, but since that was back in 2016, I can't remember exactly why we settled on one over the other. It might have been because that aligned best with our client's needs.

Since it has been this way for years, I'm not a big fan of changing it. But I'm all for making it a bit easier to customize the default value - eg. as suggested in https://github.com/skybrud/Skybrud.Umbraco.Redirects/issues/151#issuecomment-1752655093
